doc/architectural_decisions/006-where-to-put-code.md

```diff
@@ -0,0 +1,59 @@
+# Where to put code
+
+## Status
+
+Current
+
+## Context
+
+We need to decide where to draw the line between this repository's responsibility and where to put technique-specific and instrument-specific plans and devices.
+
+When we wrote initial plans on CRISP and LOQ we put everything in the instrument area and found we were duplicating a lot of stuff for common functionality ie. setting up DAEs, scanning one block, using the same callbacks etc.
+
+We decided to put all the plans in this repo, but on HIFI and RIKENFE(+ARGUS and CHRONUS) we found that this was insufficiently flexible for tweaking lots of beamline devices and plans ie. magnet tolerances and so on.
+There are different levels of use-cases where muons may need to be able to come in at a slightly lower level ie. adding lots of fits and doing back-to-back alignment scans, but these are generally specific to one instrument.
+
+## Present
+
+Tom & Jack H
+
+## Decision
+
+We should keep general and technique plans and devices in our repo, but let people write their own instrument-specific devices and plans too. These might use bits of ibex_bluesky_core, but have more flexibility. 
+
+Examples of devices and where we would put them under this model:
+
+### `ibex_bluesky_core` devices
+- `BlockRw` and `SimpleDae`: `ibex_bluesky_core` as it's completely general / useful to all beamlines
+- `ReflParameter`: in `ibex_bluesky_core.devices.reflectrometry` because it's useful across all reflectometers
+- `Danfysik`: Try to push down "special" logic (i.e. polarity switching) into the IOC so that a standard `block_rw` works. Otherwise put in `ibex_bluesky_core.devices`
+
+### `inst` scripts area devices
```

Comment:
Is this the https://github.com/ISISNeutronMuon/InstrumentScripts area?

Reply:
This is in the inst area of the config repo - perhaps we should be more clear in the ADR